integration-tests/contract-security-wrong-field-annotations.spec.ts

```diff
@@ -0,0 +1,332 @@
+import { CONFIGS } from './config';
+import { securityWrongAnnotations } from './data/security-wrong-annotations-contract';
+
+// TC-009: Check whether wrong field annotations are leading to failed transactions.
+
+CONFIGS().forEach(({ lib, rpc, setup }) => {
+  const Tezos = lib;
+
+  describe(`Test contracts to verfiy wrong field annotations are leading to failed transactions using: ${rpc}`, () => {
+    beforeEach(async (done) => {
+      await setup();
+      done();
+    });
+
+    it('Verify call Add', async () => {
+      const addition = await Tezos.contract.originate({
+        code: securityWrongAnnotations,
+        init: `0`,
+      });
+      await addition.confirmation();
+      const contractAddition = await addition.contract();
+
+      //Calls entrypoint: (pair %add (nat %valueA) (nat %valueB))
+      //Calling Michelson code: CONTRACT %add (pair (nat %valueA) (nat %valueB)) ;
+
+      try {
+      const op = await Tezos.contract
+        .at(contractAddition.address)
+        .then((contract) => {
+          return contract.methodsObject
+            .call({
+              entrypoint: 'Add',
+              param: {
+                valueA: 1,
+                valueB: 2,
+              },
+            })
+            .send();
+        })
+        .then((op) => {
+          return op.confirmation(3).then(() => op.hash);
+        })
+      } catch (ex) {
+        expect(ex).toEqual(expect.objectContaining({
+          message: expect.stringContaining('[a] Value is not a number: undefined')
```

Comment:
The annot "a" in the error doesn't match the annot of the method.
I wonder if the catch is reached or could be remove?

Reply:
the tests are to explore what happens when annotations are used in calling a contract that are not the same annotations as in the contract. 
I am not sure what the behaviour should be in these cases.